### DISCUSSION TITLE
Cursor pagination not working on bootstrap theme.

### DISCUSSION THREAD

**Discussion by nshiro** [livewire/livewire#4387] [7. Bugs (archived)] [answered]
A few days ago, cursor pagination feature was added which is great but it's not working on bootstrap theme.

Steps to reproduce

```php
class ShowUsers extends Component
{
    use WithPagination;

    protected $paginationTheme = 'bootstrap';

    public function render()
    {
        $users = User::query()
            ->orderByDesc('id')
            ->cursorPaginate(10);

        return view('livewire.show-users', [
            'users' => $users,
        ]);
    }

}

```

show-users.blade.php
```
{{ $users->links() }}

```

I get an error like so.

```
BadMethodCallException
Method Illuminate\Database\Eloquent\Collection::getPageName does not exist. (View: /___MY_PATH__/vendor/livewire/livewire/src/views/pagination/simple-bootstrap.blade.php)
```

I think the first and the fifth 'getPageName' medhod should be 'getCursorName' in simple-bootstrap.blade.php

When you add third parameter to cursorPaginate() like test does,

```php
cursorPaginate(3,'*','page')
```
It works. But other than the word 'page', it doesn't work.
I tried to fix it and send an pull request, but I had trouble with chrome dusk test part.

Thank you for reading.

**Comment by joshhanley** [comment]
@nshiro thanks for reporting!

If you already have a PR ready and are just having issues with dusk, we've just released a contribution guide here which should help with getting dusk going https://laravel-livewire.com/docs/2.x/contribution-guide

Please check it out and if you can submit your PR that would be great! 🙂

Let me know if you have any issues or need help with anything.

Hope this helps!

**Comment by nshiro** [comment]
@joshhanley Thank you, sir.

I already read the contribution guide and did what it says, but I get an error while chrome testing (I googled, but I can't find the solution yet.)
Maybe this is related to my environment (Laravel homestead).

I should keep searching for my solution, but it takes time I guess.
As much as I want to contribute, bug report is the best I can do now.

**Comment by joshhanley** [comment, marked as answer]
@nshiro no worries! Can you share a screenshot of the error so I can make sure to update the contribution guide with more troubleshooting steps?

Also don't worry about the tests then, if you can PR your fix, I will add a test in for it.

Thanks again!

**Comment by nshiro** [comment]
@joshhanley I'll send a screenshot.
I tried running test in newly installed ubuntu vagrant but I got the same error.
I'll also send an PR. Hope everything goes well.

Thank you.

![2021-12-18_00h33_14](https://user-images.githubusercontent.com/14008307/146577425-af1a0c3c-9145-4d5d-853d-28baa9fe75e2.png)

**Reply by joshhanley** [reply]
@nshiro ah interesting, I've never tried to run dusk in a VM before!

Thanks for the PR 🙂
